src/client/src/orangehrmClaimPlugin/pages/claimExpenseTypes/EditClaimExpenseType.vue

```diff
@@ -0,0 +1,166 @@
+          </oxd-grid-item>
+          <oxd-grid-item class="--offset-row-2">
+            <oxd-form-row>
+              <oxd-input-field
+                v-model="expenseTypes.description"
+                type="textarea"
+                :label="$t('general.description')"
+                :rules="rules.description"
+              />
+            </oxd-form-row>
+          </oxd-grid-item>
+          <oxd-grid-item class="--offset-row-3">
+            <div class="orangehrm-sm-field">
+              <oxd-text tag="p" class="orangehrm-sm-field-label">
+                {{ $t('general.active') }}
+              </oxd-text>
+              <oxd-switch-input v-model="expenseTypes.status" />
+            </div>
+          </oxd-grid-item>
+        </oxd-grid>
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            display-type="ghost"
+            :label="$t('general.cancel')"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {OxdSwitchInput} from '@ohrm/oxd';
+import {navigate} from '@ohrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import {shouldNotExceedCharLength} from '@ohrm/core/util/validation/rules';
+
+const initialExpenseTypes = {
+  name: '',
+  description: '',
+  status: null,
+};
+
+export default {
+  components: {
+    'oxd-switch-input': OxdSwitchInput,
+  },
+  props: {
+    id: {
+      type: Number,
+      required: true,
+    },
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/claim/expenses/types',
+    );
+    return {
+      http,
+    };
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      expenseTypes: {...initialExpenseTypes},
+      rules: {
+        description: [shouldNotExceedCharLength(1000)],
+      },
+    };
+  },
+
+  beforeMount() {
+    this.isLoading = true;
+    this.http
+      .get(this.id)
+      .then((response) => {
+        const {data} = response.data;
+        this.expenseTypes = {...data};
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  },
+
+  methods: {
+    onCancel() {
+      navigate('/claim/expenses/viewExpenseTypes');
+    },
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .update(this.id, {
+          description: this.expenseTypes.description,
+          status: this.expenseTypes.status,
+        })
+        .then(() => {
+          return this.$toast.updateSuccess();
+        })
+        .then(() => {
+          this.onCancel();
+        });
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.orangehrm-sm-field {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: $oxd-input-control-vertical-padding 0rem;
+  &-label {
+    font-weight: 700;
+    font-size: $oxd-input-control-font-size;
+  }
+}
```

Comment:
seems like repeating in claim event and expense type. Can't this move to seperate file ?